mp/src/game/client/neo/ui/neo_scoreboard.cpp

```diff
@@ -1,255 +1,774 @@
-		m_pPlayerList->AddColumnToSection(sectionID, "status", "", 0, m_iStatusWidth, hFallbackFont);
+		pPlayerList->AddColumnToSection(sectionID, "name", "", 0, m_iNameWidth - m_iAvatarWidth );
+		pPlayerList->AddColumnToSection(sectionID, "status", "", SectionedListPanel::COLUMN_IMAGE, m_iStatusWidth);
+		pPlayerList->AddColumnToSection(sectionID, "class", "Class", 0, m_iClassWidth);
+		pPlayerList->AddColumnToSection(sectionID, "rank", "Rank", 0, m_iRankWidth);
+		pPlayerList->AddColumnToSection(sectionID, "xp", "XP", 0, m_iScoreWidth);
+		pPlayerList->AddColumnToSection(sectionID, "deaths", "#PlayerDeath", 0, m_iDeathWidth );
 	}
-	else if (teamType == TYPE_SPECTATORS)
+	else if ( teamType == TYPE_SPECTATORS )
 	{
-		if (ShowAvatars())
+		pPlayerList->AddSection(sectionID, "");
+
+		// Avatars are always displayed at 32x32 regardless of resolution
+		if ( ShowAvatars() )
 		{
-			m_pPlayerList->AddColumnToSection(sectionID, "avatar", "", 0, m_iAvatarWidth);
+			pPlayerList->AddColumnToSection( sectionID, "avatar", teamNumber == TEAM_UNASSIGNED ? "Unassigned" : "Spectators", SectionedListPanel::COLUMN_IMAGE | SectionedListPanel::COLUMN_CENTER, m_iAvatarWidth );
 		}
-		m_pPlayerList->AddColumnToSection(sectionID, "name", "", 0, scheme()->GetProportionalScaledValueEx( GetScheme(), m_iNameWidth) - m_iAvatarWidth, hFallbackFont );
+		pPlayerList->AddColumnToSection(sectionID, "name", "", 0, m_iNameWidth - m_iAvatarWidth );
 	}
 
 	// set the section to have the team color
-	if (GameResources())
+	pPlayerList->SetSectionFgColor(sectionID, (teamNumber == TEAM_SPECTATOR) ? COLOR_NEO_WHITE : GameResources()->GetTeamColor(teamNumber));
+}
+
+//-----------------------------------------------------------------------------
+// Purpose: Used for sorting players
+//-----------------------------------------------------------------------------
+bool CNEOScoreBoard::StaticPlayerSortFunc(vgui::SectionedListPanel *list, int itemID1, int itemID2)
+{
+	KeyValues *it1 = list->GetItemData(itemID1);
+	KeyValues *it2 = list->GetItemData(itemID2);
+	Assert(it1 && it2);
+
+	// first compare xp
+	int v1 = it1->GetInt("xp");
+	int v2 = it2->GetInt("xp");
+	if (v1 > v2)
+		return true;
+	else if (v1 < v2)
+		return false;
+
+	// next compare deaths
+	v1 = it1->GetInt("deaths");
+	v2 = it2->GetInt("deaths");
+	if (v1 > v2)
+		return false;
+	else if (v1 < v2)
+		return true;
+
+	// the same, so compare itemID's (as a sentinel value to get deterministic sorts)
+	return itemID1 < itemID2;
+}
+
+//-----------------------------------------------------------------------------
+// Purpose: Adds a new row to the scoreboard, from the playerinfo structure
+//-----------------------------------------------------------------------------
+bool CNEOScoreBoard::GetPlayerScoreInfo(int playerIndex, KeyValues *kv)
+{
+	kv->SetInt("playerIndex", playerIndex);
+	const int neoTeam = g_PR->GetTeam(playerIndex);
```

Comment:
In other panels we check for g_PR before using it, might it be worth doing the same here and early returning if not found?